adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -781,6 +782,98 @@ public void testResiliencyTokenReturnExtendedLifetimeOnwithoutRetry() throws Pac
         }
     }
     
+    @SmallTest
+    public void testVerifyManifestPermission_valid() throws InterruptedException, PackageManager.NameNotFoundException {
+        final FileMockContext mockContext = createMockContext();
+        AuthenticationSettings.INSTANCE.setUseBroker(true);
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false);
+        final TestAuthCallback callback = new TestAuthCallback();
+        authContext.acquireToken(Mockito.mock(Activity.class), "resource", "clientid", authContext.getRedirectUriForBroker(),
+                "loginHint", callback);
+
+        assertNull(callback.getCallbackException());
+    }
+
+    public void testVerifyManifestPermission_missingPermission_MANAGE_ACCOUNTS() throws InterruptedException, PackageManager.NameNotFoundException {
+        final FileMockContext mockContext = createMockContext();
+        mockContext.removePermission("android.permission.MANAGE_ACCOUNTS");
+        AuthenticationSettings.INSTANCE.setUseBroker(true);
+        
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false);
+        final TestAuthCallback callback = new TestAuthCallback();
+        authContext.acquireToken(Mockito.mock(Activity.class), "resource", "clientid", authContext.getRedirectUriForBroker(),
+                "loginHint", callback);
+
+        assertNotNull(callback.getCallbackException());
+        assertTrue(callback.getCallbackException() instanceof UsageAuthenticationException);
+        final UsageAuthenticationException usageAuthenticationException
+                = (UsageAuthenticationException) callback.getCallbackException();
+        assertTrue(usageAuthenticationException.getMessage().contains("MANAGE_ACCOUNTS"));
+        assertEquals(ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING, usageAuthenticationException.getCode());
+    }
+
+    public void testVerifyManifestPermission_missingPermission_USE_CREDENTIALS() throws InterruptedException, PackageManager.NameNotFoundException {
```

Comment:
this is not the naming convention. We did the work to make sure that our code is clean from the two static analysis tools, even though tests files are excluded from the check, I think we should make sure we do our best to not introduce new code in test that is not compliant with the rule sets.